### PR TITLE
HIVE-27289: Check for proxy hosts with subnet in IP results in exception

### PR DESCRIPTION
**Pull request by nrg4878**
### What changes were proposed in this pull request?
In MetastoreServerUtils.checkUserHasHostProxyPrivileges(), the check is to see if hadoop.proxyuser* settings in core-site.xml allow this user to execute this. One of the arguments to this method is an IpAddress that it is being executed on, which is fetched from HMSHandler.getIpAddress(). However, when running schematool, this is never is set during initialization. In this case, because it is not HMS server, the SaslServer is not initialized. 
When the core-site.xml has CIDR configurations, SubnetInfo.inRange() check throws an exception when the ip we are checking on in EMPTY.
So in case of schematool, try to use the hostaddress before resorting to empty ip address.

### Why are the changes needed?
HIVE-27289. To avoid exception during upgrades.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Manually.

### CODE REVIEW

File: jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/TeradataDatabaseAccessor.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+package org.apache.hive.storage.jdbc.dao;
+
+/**
+ * Teradata specific data accessor. This is needed because Teradata JDBC drivers do not support generic LIMIT and OFFSET
+ * escape functions
+ */
+public class TeradataDatabaseAccessor extends GenericJdbcDatabaseAccessor {
+
+  @Override
```
Comment (dengzhhu653) [inline jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/TeradataDatabaseAccessor.java:23]: nit: this file seems unrelated to the problem, should we create another jira for the feature? I think it makes better for porting or tracking the issue.

**Comment by dengzhhu653** [comment]
+1, a minor comment, others look good to me

File: jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/TeradataDatabaseAccessor.java (hunk 1 above)
Comment (nrg4878) [reply jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/TeradataDatabaseAccessor.java:23]: Not a nit. There should only be one file in this change, not sure how others got pulled in. Let me fix that.

Comment (nrg4878) [reply jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/TeradataDatabaseAccessor.java]: Fixed.

**Review by dengzhhu653 — APPROVED** [review]
